### PR TITLE
API-8656: adjusted SDK to v3.3

### PR DESCRIPTION
**Pull request by grz-gajda**
After command `npm run docs` I've got 190 files changed so I am not sure should I run this or not.

### CODE REVIEW

File: src/agent/web.ts [hunk 1]
```diff
@@ -416,4 +417,34 @@ export default class Web extends WebAPI {
   async listAgentsForTransfer(chat_id: string): Promise<AgentForTransfer[]> {
     return this.send("list_agents_for_transfer", { chat_id });
   }
+
+  /**
+   * Marks a customer as followed. As a result, the requester (an agent) will receive
+   * the info about all the changes related to that customer via pushes.
+   * @param id - ID of customer
```
Comment (janowsiany) [inline src/agent/web.ts:424]: opt:
```suggestion
   * @param id - ID of customer to follow
```
This change is to align with other param descriptions

File: src/agent/web.ts [hunk 2]
```diff
@@ -416,4 +417,34 @@ export default class Web extends WebAPI {
...
+   */
+  async followCustomer(id: string): Promise<EmptyResponse> {
+    return this.send("follow_customer", { id });
+  }
+
+  /**
+   * Removes the agent from the list of customer's followers.
+   * @param id - ID of customer
```
Comment (janowsiany) [inline src/agent/web.ts:432]:
```suggestion
   * @param id - ID of customer to unfollow
```
same as above

File: src/agent/web.ts [hunk 3]
```diff
@@ -416,4 +417,34 @@ export default class Web extends WebAPI {
...
+
+  /**
+   * Returns the current routing status of each agent.
+   * @param groupIds - Array of group IDs.
+   */
+  async listRoutingStatuses(groupIds?: number[]): Promise<SetRoutingStatusResponse[]> {
+    const filters: { group_ids?: number[] } = {};
+    if (groupIds && groupIds?.length > 0) {
+      filters.group_ids = groupIds;
+    }
```
Comment (janowsiany) [inline src/agent/web.ts:446]: i think we do not put such logic here in the sdk, check other similar places

File: src/agent/web.ts [hunk 4]
```diff
@@ -416,4 +417,34 @@ export default class Web extends WebAPI {
...
+   */
+  async unfollowCustomer(id: string): Promise<EmptyResponse> {
+    return this.send("unfollow_customer", { id });
+  }
+
+  /**
+   * Returns the current routing status of each agent.
+   * @param groupIds - Array of group IDs.
+   */
+  async listRoutingStatuses(groupIds?: number[]): Promise<SetRoutingStatusResponse[]> {
```
Comment (janowsiany) [inline src/agent/web.ts:442]: i think we use snake_case to name params, check other similar places for reference

File: src/agent/web.ts [hunk 5]
```diff
@@ -416,4 +417,34 @@ export default class Web extends WebAPI {
...
+   * Removes the agent from the list of customer's followers.
+   * @param id - ID of customer
+   */
+  async unfollowCustomer(id: string): Promise<EmptyResponse> {
+    return this.send("unfollow_customer", { id });
+  }
+
+  /**
+   * Returns the current routing status of each agent.
+   * @param groupIds - Array of group IDs.
```
Comment (janowsiany) [inline src/agent/web.ts:440]:
```suggestion
   * @param group_ids - groups to list agent routing statuses from
```

File: src/configuration/index.ts [hunk 6]
```diff
@@ -373,4 +377,39 @@ export default class ConfigurationAPI extends WebAPI {
   async getLicenseWebhooksState(client_id?: string): Promise<WebhooksState> {
     return this.send("get_license_webhooks_state", { client_id });
   }
+
+  /**
+   * Creates an auto access data structure, which is a set of conditions for the tracking URL and geolocation of a customer.
+   * @param opts - options for
```
Comment (janowsiany) [inline src/configuration/index.ts:383]: something is missing in the description?

File: src/configuration/index.ts [hunk 7]
```diff
@@ -373,4 +377,39 @@ export default class ConfigurationAPI extends WebAPI {
...
+  /**
+   * Returns all existing auto access data structures.
+   */
+  async listAutoAccesses(): Promise<AutoAccess[]> {
+    return this.send("list_auto_accesses", {});
+  }
+
+  /**
+   * Deletes an existing auto access data structure specified by its ID.
+   * @param id - auto access ID
```
Comment (janowsiany) [inline src/configuration/index.ts:398]: please follow convention in param description

File: src/configuration/index.ts [hunk 8]
```diff
@@ -373,4 +377,39 @@ export default class ConfigurationAPI extends WebAPI {
...
+   */
+  async deleteAutoAccess(id: string): Promise<EmptyResponse> {
+    return this.send("delete_auto_access", { id });
+  }
+
+  /**
+   * Updates an existing auto access. Only specified fields are updated (overwritten), leaving unspecified fields unchanged.
+   * @param id - 	ID of the auto access to modify
```
Comment (janowsiany) [inline src/configuration/index.ts:406]: is spacing correct here?

File: src/customer/web.ts [hunk 9]
```diff
@@ -358,6 +352,40 @@ export default class Web extends WebAPI {
...
+   * @param opts - properties used to groups' filtering
+   */
+  async getDynamicConfiguration(opts: GetDynamicConfigurationRequest = {}): Promise<GetDynamicConfigurationResponse> {
+    return this.send("get_dynamic_configuration", opts);
+  }
+
+  /**
+   * Returns the configuration of a given group in a given version.
+   * @param group_id
+   * @param version
```
Comment (janowsiany) [inline src/customer/web.ts:369]: do we have a description for that?

File: src/customer/web.ts [hunk 10]
```diff
@@ -358,6 +352,40 @@ export default class Web extends WebAPI {
...
+    return this.send("request_email_verification", { callback_uri });
+  }
+
+  /**
+   * Returns the dynamic configuration of a given group.
+   * @param opts - properties used to groups' filtering
+   */
+  async getDynamicConfiguration(opts: GetDynamicConfigurationRequest = {}): Promise<GetDynamicConfigurationResponse> {
+    return this.send("get_dynamic_configuration", opts);
+  }
```
Comment (janowsiany) [inline src/customer/web.ts:364]:
```suggestion
  async getDynamicConfiguration(opts?: GetDynamicConfigurationRequest): Promise<GetDynamicConfigurationResponse> {
    return this.send("get_dynamic_configuration", opts || {});
  }
```
i think that in other similar places we do it the way presented above, not sure which is better but I prefer to keep it consistent, wdyt?

File: src/customer/web.ts [hunk 11]
```diff
@@ -358,6 +352,40 @@ export default class Web extends WebAPI {
...
+    return this.send("get_configuration", {
+      group_id,
+      version,
+    });
+  }
+
+  /**
+   * Returns the localization of a given language and group in a given version.
+   * @param group_id - The ID of the group that you want to get a localization for
+   * @param language - 	The language that you want to get a localization for
```
Comment (janowsiany) [inline src/customer/web.ts:381]: spacing is correct here?

**Review by janowsiany — CHANGES_REQUESTED** [review]

File: src/agent/web.ts (hunk 3 above)
Comment (grz-gajda) [reply src/agent/web.ts]: Ok, I found example:

```ts
  async listAgents(group_ids?: number[], fields?: string[]): Promise<Agent[]> {
    return this.send("list_agents", { fields, filters: { group_ids } });
  }
```

Changed to remain consistent with the rest of the codebase.

File: src/agent/web.ts (hunk 4 above)
Comment (grz-gajda) [reply src/agent/web.ts]: That's right, changed.

File: src/customer/web.ts (hunk 10 above)
Comment (grz-gajda) [reply src/customer/web.ts:364]: Sure, changed to be consistent.

File: src/agent/web.ts (hunk 1 above)
Comment (grz-gajda) [reply src/agent/web.ts]: Fixed. Completed the description for parameters.

File: src/configuration/index.ts (hunk 6 above)
Comment (grz-gajda) [reply src/configuration/index.ts]: Fixed. Completed the description for parameters.

File: src/configuration/index.ts (hunk 7 above)
Comment (grz-gajda) [reply src/configuration/index.ts]: Fixed. Completed the description for parameters.

File: src/configuration/index.ts (hunk 8 above)
Comment (grz-gajda) [reply src/configuration/index.ts]: Fixed.

File: src/customer/web.ts (hunk 9 above)
Comment (grz-gajda) [reply src/customer/web.ts]: Fixed. Completed the description for parameters.

File: src/customer/web.ts (hunk 11 above)
Comment (grz-gajda) [reply src/customer/web.ts]: Fixed.

**Review by janowsiany — CHANGES_REQUESTED** [review]

File: src/customer/structures.ts [hunk 12]
```diff
@@ -179,6 +182,38 @@ export interface LoginResponse {
...
+  localization_version: string;
+  language: string;
+}
+
+export interface GroupConfiguration {
+  buttons: {
+    id: string;
+    type: string;
+    online_value: string;
+    offline_valie: string;
```
Comment (kogierman) [inline src/customer/structures.ts:206]: typo: valie -> value

File: src/customer/structures.ts [hunk 13]
```diff
@@ -150,7 +153,7 @@ export interface LoginRequest {
   customer_side_storage?: object;
   group_id?: number;
   referrer?: string;
-  pushes?: { [key: string]: Pushes[]};
+  pushes?: { [key: string]: Pushes[] };
```
Comment (kogierman) [inline src/customer/structures.ts:156]: this definition is inconsistent with Agent part ;)

also, did you decide to ditch the idea of push - version mapping?

**Review by kogierman — CHANGES_REQUESTED** [review]

File: src/agent/web.ts (hunk 4 above)
Comment (kogierman) [reply src/agent/web.ts]: yup, in most places we use function param names to directly construct req payloads, which require snake_case, therefore snake_case is chosen here (camelCase everywhere else)

File: src/customer/structures.ts (hunk 13 above)
Comment (grz-gajda) [reply src/customer/structures.ts:156]: Yes, I think versioning pushes by Typescript's types will be too overhead without any gain (excluding autocomplete in some editors) and it'll increase the cost of maintaining the code.

File: src/customer/structures.ts (hunk 12 above)
Comment (grz-gajda) [reply src/customer/structures.ts:206]: Fixed

File: src/customer/web.ts [hunk 14]
```diff
@@ -358,6 +352,40 @@ export default class Web extends WebAPI {
    * @param callback_uri - URI to send webhook when customer confirms identity
    */
   async requestEmailVerification(callback_uri: string): Promise<EmptyResponse> {
-	  return this.send("request_email_verification", { callback_uri });
+    return this.send("request_email_verification", { callback_uri });
+  }
+
+  /**
+   * Returns the dynamic configuration of a given group.
+   * @param opts - properties used to groups' filtering
```
[the remark this answers is not in the document]
Comment (grz-gajda) [reply src/customer/web.ts:360]: I don't have a better idea, I'll replaced it with
> properties used to find matching group

**Review by janowsiany — APPROVED** [review]

**Review by kogierman — APPROVED** [review]

**Pull request merged**